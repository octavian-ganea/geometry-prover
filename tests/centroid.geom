Description : intersection of altitudes in a triangle is a single point - the ortocenter

Hypothesis:

Let an arbitrary circle Circ(O ,radix)

Let an semiarbitrary point A situated on circle Circ
Let an semiarbitrary point B situated on circle Circ
Let an semiarbitrary point C situated on circle Circ

Let a 1-equation deterministic point A1 situated on midpoint of B C
Let a 1-equation deterministic point C1 situated on midpoint of B A
Let a 1-equation deterministic point B1 situated on midpoint of A C

Conclusion:
concurrent line A A1 line B B1 line C C1
