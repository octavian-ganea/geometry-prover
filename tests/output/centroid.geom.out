====== Parsing .... 
Description : intersection of altitudes in a triangle is a single point - the ortocenter 

Hypothesis:

Let an arbitrary circle Circ(O,radix)

Let a semiarbitrary point A situated on circle Circ
Let a semiarbitrary point B situated on circle Circ
Let a semiarbitrary point C situated on circle Circ

Let a 1-equation deterministic point A1 situated on midpoint of B C
Let a 1-equation deterministic point C1 situated on midpoint of B A
Let a 1-equation deterministic point B1 situated on midpoint of A C
Let a 2-equations deterministic point X13 situated on line A-A1 and line B-B1

Conclusion: points X13, C and C1 are collinear

====== Parse OK 

====== Proof of the problem:

======= Generating equations for each point ... 

========= List of arbitrary points:


========= List of semi-arbitrary points:

A is defined by equation: a * a__ - 1 = 0
After substitutions and simplifications:
 a__ = 1  / a

B is defined by equation: b * b__ - 1 = 0
After substitutions and simplifications:
 b__ = 1  / b

C is defined by equation: c * c__ - 1 = 0
After substitutions and simplifications:
 c__ = 1  / c

========= List of deterministic points:

A1 is a deterministic point defined by equations (After substitutions and simplifications):
 
 a1  =  b + c  / 2 

 a1__  =  c + b  / 2 * b * c 


C1 is a deterministic point defined by equations (After substitutions and simplifications):
 
 c1  =  b + a  / 2 

 c1__  =  a + b  / 2 * a * b 


B1 is a deterministic point defined by equations (After substitutions and simplifications):
 
 b1  =  a + c  / 2 

 b1__  =  c + a  / 2 * a * c 


X13 is defined by equation: a__ * x13 + a * a1__ + a1 * x13__ - a1__ * x13 - a * x13__ - a1 * a__ = 0
         and by equation: b__ * x13 + b * b1__ + b1 * x13__ - b1__ * x13 - b * x13__ - b1 * b__ = 0
After substitutions and simplifications:
 x13  =  c + b + a  / 3
 x13__  =  a * b + a * c + b * c  / 3 * a * b * c 

=========== Equations were generated successfully ========

======= The conclusion may be correct only unde the following restrictions:
 -- points B and A must be different
 -- points C and A must be different
 -- points C and B must be different

====== Trying to solve the goal of the problem ...
Conclusion is equivalent to equation: 
c__ * x13 + c * c1__ + c1 * x13__ - c1__ * x13 - c * x13__ - c1 * c__ = 0

Attempt to prove it by substitutions and simplifications ....
 
  !!!!!!!!!!!!!!!!!!!!!
  !!!!!! SUCCESS !!!!!!
  !!!!!!!!!!!!!!!!!!!!!

===== Problem is correct. Proof ends here. QED ====

